registry/unit.go

```diff
@@ -62,8 +64,46 @@ func (r *EtcdRegistry) getUnitByHash(hash unit.Hash) *unit.UnitFile {
 		}
 		return nil
 	}
+	return r.unitFromEtcdNode(hash, resp.Node)
+}
+
+// getAllUnitsHashMap retrieves from the Registry all Units and returns a map of hash to Unit
+func (r *EtcdRegistry) getAllUnitsHashMap() (map[string]*unit.UnitFile, error) {
+	key := r.prefixed(unitPrefix)
+	opts := &etcd.GetOptions{
+		Recursive: true,
```

Comment:
Will do. But out of curiosity, what is the canonical etcd Go client?
https://github.com/coreos/go-etcd
or https://github.com/coreos/etcd/client?